examples/pytorch/graphsage/train_full.py

```diff
@@ -167,5 +83,42 @@ def main(args):
                         help="Aggregator type: mean/gcn/pool/lstm")
     args = parser.parse_args()
     print(args)
+    
+    # load and preprocess dataset
+    data = load_data(args)
+    g = data[0]
+    features = g.ndata['feat']
+    labels = g.ndata['label']
+    train_mask = g.ndata['train_mask']
+    val_mask = g.ndata['val_mask']
+    test_mask = g.ndata['test_mask']
+    in_feats = features.shape[1]
+    n_classes = data.num_classes
+    n_edges = g.number_of_edges()
+    print("""----Data statistics------'
+      #Edges %d
+      #Classes %d
+      #Train samples %d
+      #Val samples %d
+      #Test samples %d""" %
+          (n_edges, n_classes,
+           train_mask.int().sum().item(),
+           val_mask.int().sum().item(),
+           test_mask.int().sum().item()))
 
-    main(args)
+    if args.gpu < 0:
+        device = torch.device('cpu')
+    else:
+        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+        features = features.to(device)
+        labels = labels.to(device)
+        g = g.to(device)
+        
+    train_nid = train_mask.nonzero().squeeze().to(device)
+    val_nid = val_mask.nonzero().squeeze().to(device)
+    test_nid = test_mask.nonzero().squeeze().to(device)
```

Comment:
I'm curious if `squeeze` is always used when `train_mask`/`val_mask`/`test_mask` are used.

Reply:
Not for all examples.  But I think yes in graphsage. 
We should use "squeeze" instead of full train_mask as much as possible for performance, especially when graph size scales up